pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -0,0 +1,204 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topologymanager
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
+)
+
+//Manager interface provides methods for Kubelet to manage pod topology hints
+type Manager interface {
+	//Manager implements pod admit handler interface
+	lifecycle.PodAdmitHandler
+	//Adds a hint provider to manager to indicate the hint provider
+	//wants to be consoluted when making topology hints
+	AddHintProvider(HintProvider)
+	//Adds pod to Manager for tracking
+	AddContainer(pod *v1.Pod, containerID string) error
+	//Removes pod from Manager tracking
+	RemoveContainer(containerID string) error
+	//Interface for storing pod topology hints
+	Store
+}
+
+//TopologyHint is a struct containing Socket Mask for a Pod
+type TopologyHint struct {
+	SocketMask socketmask.SocketMask
+}
+
+type manager struct {
+	//The list of components registered with the Manager
+	hintProviders []HintProvider
+	//List of Containers and their Topology Allocations
+	podTopologyHints map[string]containers
+	podMap           map[string]string
+	//Topology Manager Policy
+	policy Policy
+}
+
+//HintProvider interface is to be implemented by Hint Providers
+type HintProvider interface {
+	GetTopologyHints(pod v1.Pod, container v1.Container) ([]TopologyHint, bool)
+}
+
+//Store interface is to allow Hint Providers to retrieve pod affinity
+type Store interface {
+	GetAffinity(podUID string, containerName string) TopologyHint
+}
+
+type containers map[string]TopologyHint
+
+var _ Manager = &manager{}
+
+type policyName string
+
+//NewManager creates a new TopologyManager based on provided policy
+func NewManager(topologyPolicyName string) Manager {
+	klog.Infof("[topologymanager] Creating topology manager with %s policy", topologyPolicyName)
+	var policy Policy
+
+	switch policyName(topologyPolicyName) {
+
+	case PolicyPreferred:
+		policy = NewPreferredPolicy()
+
+	case PolicyStrict:
+		policy = NewStrictPolicy()
+
+	default:
+		klog.Errorf("[topologymanager] Unknow policy %s, using default policy %s", topologyPolicyName, PolicyPreferred)
+		policy = NewPreferredPolicy()
+	}
+
+	var hp []HintProvider
+	pnh := make(map[string]containers)
+	pm := make(map[string]string)
+	manager := &manager{
+		hintProviders:    hp,
+		podTopologyHints: pnh,
+		podMap:           pm,
+		policy:           policy,
+	}
+
+	return manager
+}
+
+func (m *manager) GetAffinity(podUID string, containerName string) TopologyHint {
+	return m.podTopologyHints[podUID][containerName]
+}
+
+func (m *manager) calculateTopologyAffinity(pod v1.Pod, container v1.Container) (TopologyHint, bool) {
```

Comment:
Would a better name for this function just be `calculateAffinity` for symmetry with `GetAffinity` above?